### DISCUSSION TITLE
Mod select feature request

### DISCUSSION THREAD

**Discussion by Theighlin** [ppy/osu#18172] [Ideas] [answered]
Sorry for the title, couldn't express it much better.

The previous mod select overlay had a couple of features that i liked, though i feel they were more a result of the working of the overlay itself rather than intended.

The first one is the one I used more and feel like it can be added back as a behaviour, and it is being able to start gameplay from the overlay. When i came accross a song i liked in the listing, i would usually open the overlay, select approach different, maybe tweak it and jump into the song. Now you need to go back 2 times to play after setting up the mods, which is unfortunate. I saw the osu cookie doesn't seem to overlap the mod sections, so i thought maybe it could stay on top as before? (the settings section is easy to scroll and it's difficult that it reached the far right of the screen).
It's true that stable doesn't do this, but you also couldn't change settings of mods, half as annoying to me.

https://user-images.githubusercontent.com/53081750/167385822-871cc4ce-1e1e-4ebb-86ee-72cc1a6bb945.mp4

https://user-images.githubusercontent.com/53081750/167385866-52720083-8c84-45e8-8b6b-7cd055cdcfa8.mp4

The second one is just being able to see the difference in star rating change on the info wedge as i'm selecting mods, not something stable does either, and less important, but was a nice to have imo.

https://user-images.githubusercontent.com/53081750/167385844-76310a47-94f5-4dcb-972b-b36fd2c46007.mp4

https://user-images.githubusercontent.com/53081750/167385880-0920edf7-9f7b-4e41-be6e-0d36d407b586.mp4

**Comment by peppy** [comment]
You can press enter twice to start the beatmap from the mod select screen. I tried making it require a single press, but it felt bad. I think you will learn to double-tap.

Can agree about difficulty. We probably want it to show inline, next to the multiplier display.

**Reply by Nepp0** [reply]
i dont think its worth to create another thread for this but i would also like to point out that some previously usable hotkeys dont seem to work with the new mod select, like "1" for reset all mods and "2" for closing mod select screen.

**Reply by Moltanicaa** [reply]
It would also be nice to add customization of hotkeys for this section. I really miss the function that the "1" key disables all selected mods.

**Reply by bdach** [reply]
Someone else opened a separate discussion about it here: https://github.com/ppy/osu/discussions/18186 - can you go there and see what you think about my suggestion there?

**Reply by ikkue** [reply]
The new mod select screen design covers pretty much the entire screen anyways, so I think it'd be nice to display necessary information, or info that can be affected by applying mods (Star Rating, Map Length, AR, HP, OD, CS) in the entire space next to the difficulty multiplier so there's essentially no blank space left.

**Reply by bdach** [reply]
This was what I had in mind but haven't yet explored how that's going to go.

**Reply by frenzibyte** [reply]
Going to mention that part is tracked in https://github.com/ppy/osu/discussions/18189 instead.

**Comment by frenzibyte** [comment, marked as answer]
Going to mark this as a duplicate of https://github.com/ppy/osu/discussions/18186 for the first point and https://github.com/ppy/osu/discussions/18189 for the second point. Please consider opening one discussion thread per proposal to make things easier next time.
